flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -202,6 +202,29 @@ genProdOrSum(FN func, FD funcDim, mlir::Type resultType,
                     args[1], mask, rank);
 }
 
+/// Process calls to Product, Sum intrinsic functions
+template <typename FN>
+static fir::ExtendedValue
+genDotProd(FN func, mlir::Type resultType,
+           Fortran::lower::FirOpBuilder &builder, mlir::Location loc,
+           Fortran::lower::StatementContext *stmtCtx, llvm::StringRef errMsg,
+           llvm::ArrayRef<fir::ExtendedValue> args) {
+
+  assert(args.size() == 2);
+
+  // Handle required vector arguments
+  fir::BoxValue arryTmpA = builder.createBox(loc, args[0]);
+  mlir::Value arrayA = fir::getBase(arryTmpA);
+  fir::BoxValue arryTmpB = builder.createBox(loc, args[1]);
+  mlir::Value arrayB = fir::getBase(arryTmpB);
+  int rank = arryTmpA.rank();
+  assert(rank == 1);
```

Comment:
I suggest adding an assert for the rank of arryTmpB too.

Reply:
It was suggested to me in my PR #916 that we do not have to check the rank here. It should already be done in the front-end. So, you can probably get rid of the assert(rank == 1).